nest_elephant_tvb/translation/FatEndPoint.py

```diff
@@ -0,0 +1,196 @@
+from mpi4py import MPI
+import pathlib
+
+def make_connections(path_to_files_receive, path_to_files_send, logger_master):
+    '''
+    Fat end point, here: connections to nest and tvb.
+    '''
+    use_mpi = True
+    # init MPI
+    if use_mpi:
+        comm = MPI.COMM_WORLD  # INTRA communicator
+        info = MPI.INFO_NULL
+        root=0
+    else:
+        comm = None
+    ### nest connection
+    comm_receiver, port_receive = nest_connection(comm, root, info, logger_master, path_to_files_receive)
+    ### tvb connecction
+    comm_sender, port_send = tvb_connection(comm, root, info, logger_master, path_to_files_send)
+    logger_master.info('Connections made, starting translation...')
+    
+    return comm_receiver, port_receive, comm_sender, port_send
+
+
+def nest_connection(comm, root, info, logger_master, path_to_files_receive):
+    '''
+    MPI inter communicator to NEST
+    
+    '''
+    logger_master.info('Translate Receive: before open_port')
+    if comm.Get_rank() == 0:
+        ### Connection to simulation (incoming data)
+        port_receive = MPI.Open_port(info)
+        logger_master.info('Translate Receive: after open_port: '+port_receive)
+        # Write file configuration of the port
+        fport = open(path_to_files_receive, "w+")
+        fport.write(port_receive)
+        fport.close()
+        pathlib.Path(path_to_files_receive+'.unlock').touch()
+        logger_master.info('Translate Receive: path_file: ' + path_to_files_receive)
+    else:
+        port_receive = None
+    
+    # broadcast port info, accept connection on all ranks!
+    # necessary to avoid problems with information about the mpi rank in open port file.
+    port_receive = comm.bcast(port_receive,root)
```

Comment:
Reuse of the same variable. What does this bcast do to the content of port_receive?

Reply:
This bcast, as the comment says, avoids problems when accepting MPI connections with multiple ranks.
Some Machines (local computers, supercomputers, ...) store RANK information in the files.
This would then create multiple connections. To avoid this, the 'if statement' above only opens the Port on rank 0 and then distributes this information to all other ranks.

Variable name must be the same on all ranks.

Reply:
I leave it as unresolved for the moment, we can discuss and add more information in the comment.